pkg/graph/schema.graphql

```diff
@@ -1,2301 +1,2326 @@
+    modifiedBy: String
+    modifiedDts: Time
+}
+
+input OperationalSolutionChanges @goModel(model: "map[string]interface{}"){
+    needed: Boolean
+    pocName: String
+    pocEmail: String
+    mustStartDts: Time
+    mustFinishDts: Time
+    status: OpSolutionStatus
+}
+
+input PlanDocumentSolutionLinkInput {
+id: UUID!
+modelPlanID: UUID!
+solutionID: Int!
+documentID: UUID!
+}
+
+type PlanDocumentSolutionLink {
+id: UUID!
+modelPlanID: UUID!
+solutionID: Int!
+documentID: UUID!
+createdBy: String!
+createdDts: Time!
+modifiedBy: String
+modifiedDts: Time
+}
+
+"""
+Query definition for the schema
+"""
+type Query {
+  currentUser: CurrentUser!
+  modelPlan(id: UUID!): ModelPlan!
+  planDocument(id: UUID!): PlanDocument!
+  modelPlanCollection(includeAll: Boolean!): [ModelPlan!]!
+  existingModelCollection: [ExistingModel!]!
+  cedarPersonsByCommonName(commonName: String!): [UserInfo!]!
+  planCollaboratorByID(id: UUID!): PlanCollaborator!
+  taskListSectionLocks(modelPlanID: UUID!): [TaskListSectionLockStatus!]!
+  planPayments(id: UUID!): PlanPayments!
+  ndaInfo: NDAInfo!
+  crTdl(id: UUID!): PlanCrTdl!
+  operationalSolutions(operationalNeedID: UUID!, includeNotNeeded: Boolean! = false): [OperationalSolution!]!
+  operationalSolution(id: UUID!): OperationalSolution!
+  operationalNeed(id: UUID!): OperationalNeed!
+  auditChanges(tableName: String!, primaryKey: UUID!): [AuditChange!]!
+  possibleOperationalNeeds: [PossibleOperationalNeed!]!
+  planDocumentSolutionLinks(modelPlanID: UUID!): [PlanDocumentSolutionLink!]!
+}
+
+"""
+Mutations definition for the schema
+"""
+type Mutation {
+createModelPlan(modelName: String!): ModelPlan!
+@hasRole(role: MINT_USER)
+
+updateModelPlan(id: UUID!, changes: ModelPlanChanges!): ModelPlan!
+@hasRole(role: MINT_USER)
+
+createPlanCollaborator(input: PlanCollaboratorCreateInput!): PlanCollaborator!
+@hasRole(role: MINT_USER)
+
+updatePlanCollaborator(id: UUID!, newRole: TeamRole!): PlanCollaborator!
+@hasRole(role: MINT_USER)
+
+deletePlanCollaborator(id: UUID!): PlanCollaborator!
+@hasRole(role: MINT_USER)
+
+updatePlanBasics(id: UUID!, changes: PlanBasicsChanges!): PlanBasics!
+@hasRole(role: MINT_USER)
+
+updatePlanGeneralCharacteristics(id: UUID!, changes: PlanGeneralCharacteristicsChanges!): PlanGeneralCharacteristics!
+@hasRole(role: MINT_USER)
+
+updatePlanBeneficiaries(id: UUID!, changes: PlanBeneficiariesChanges!): PlanBeneficiaries!
+@hasRole(role: MINT_USER)
+
+updatePlanParticipantsAndProviders(id: UUID!, changes: PlanParticipantsAndProvidersChanges!): PlanParticipantsAndProviders!
+@hasRole(role: MINT_USER)
+
+updatePlanItTools(id: UUID!, changes:PlanITToolsChanges!): PlanITTools!
+@hasRole(role: MINT_USER)
+
+updatePlanOpsEvalAndLearning(id: UUID!, changes: PlanOpsEvalAndLearningChanges!): PlanOpsEvalAndLearning!
+@hasRole(role: MINT_USER)
+
+uploadNewPlanDocument(input: PlanDocumentInput!): PlanDocument!
+@hasRole(role: MINT_USER)
+
+deletePlanDocument(id: UUID!): Int!

```

Comment:
We should probably add functionality to this mutation to delete the plan document links as well.